modules/network/src/main/java/org/apache/ignite/internal/network/message/ClusterNodeMessage.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.message;
+
+import static org.apache.ignite.internal.network.NetworkMessageTypes.CLUSTER_NODE_MESSAGE;
+
+import java.io.Serializable;
+import org.apache.ignite.internal.network.ClusterNodeImpl;
+import org.apache.ignite.internal.network.NetworkMessage;
+import org.apache.ignite.internal.network.annotations.Transferable;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.NetworkAddress;
+
+/**
+ * {@link ClusterNode} as a network message class.
+ */
+@Transferable(CLUSTER_NODE_MESSAGE)
+public interface ClusterNodeMessage extends NetworkMessage, Serializable {
```

Comment:
We already have a `ClusterNodeMessage`, can we reuse it?

Reply:
It's in another module and it is specific to the CMG: take a look at its 3 last methods. I also considered a possibility to create a base interface in the network module and extend it in the CMG modules with those 3 methods, but this does not seem to give a lot of gain (it allows us to reduce duplication by removing 4 lines defining id/name/host/port), but it introduces a dependency which usually bites afterwards. That's why it seemed to me to just copy the interface and remove the 3 last methods from it.

Reply:
Those three last methods are nullable, can we simply pass nulls?

Reply:
This looks like an abuse, to be honest. And that message is in another module.